src/emc/usr_intf/stepconf/build_HAL.py

```diff
@@ -23,393 +23,423 @@
 import os
 import time
 import shutil
+from stepconf.definitions import *
 
 class HAL:
     def __init__(self,app):
-        # access to:
-        self.d = app.d  # collected data
-        global SIG
-        SIG = app._p    # private data (signal names)
-        self.a = app    # The parent, stepconf
+		# access to:
+		self.d = app.d  # collected data
+		self.a = app    # The parent, stepconf
 
     def write_halfile(self, base):
-        inputs = self.a.build_input_set()
```

Comment:
Sorry, I guess it's Geany editor fault. It does not behave very well with python indentations if there are spaces and tabulations mixed together. I have uniformed the indentations in the file.